### ISSUE TITLE
crafting issues

### ISSUE THREAD

**Issue by kingvor** [tsunamayo/Starship-EVO#4974]
### Short Description of the issue:
this is a list of features i noticed not working with the assembler
1: assembler  doesnt pull all the neceseary components from cargo container
2: assembler doesnt push components and blocks to cargo containers
...

### Steps to reproduce the issue:
place a assembler set up with pipes and cargo and a assembler tweak the settings of the assembler and try it, you will see it isnt pulling compnenst and not pushing finished blocks to cargo containers

...

### Environment
- Starship EVO version: 22w42a
- Graphics Card: 

### Additional information:

_(Screenshots, Videos, Error logs, Related issues)_

Please also include the `Player.log` file. You can find the log file here: `%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\`

If you issue concerns a particular design please include any blueprint that could help to reproduce the issue.
Blueprint folder is over at:
`%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\Save_Data\Blueprints\`
You will need to zip the folder to upload it. 
To do so Right click and choose [Send to -> compressed zip file] then drag the .zip onto your report to upload it.

**Comment by tsunamayo** [comment]
@kingvor Hi, I can reproduce the cannot push component issue (bricks are fine), but in my test the pull was working fine. Are you sure it is also not working? The assembler will only pull components that are needed. Thanks!

**Comment by kingvor** [comment]
> @kingvor Hi, I can reproduce the cannot push component issue (bricks are fine), but in my test the pull was working fine. Are you sure it is also not working? The assembler will only pull components that are needed. Thanks!

yes i am sure, i tested it and i had to put in the componets in the assembler for it to work, it would only pull one item then stop
